xcodeproj/internal/xcodeproj_runner.bzl

```diff
@@ -111,17 +220,28 @@ def _write_runner(
         config,
         extra_flags_bazelrc,
         extra_generator_flags,
-        generator_label,
+        generator_build_file,
+        generator_defs_bzl,
         is_fixture,
         project_name,
         runner_label,
+        schemes_json,
         template,
         xcode_version,
         xcodeproj_bazelrc):
     output = actions.declare_file("{}-runner.sh".format(name))
 
     is_bazel_6 = hasattr(apple_common, "link_multi_arch_static_library")
 
+    generator_package_directory = (
+        ".rules_xcodeproj/" +
+        paths.join(
+            paths.dirname(generator_build_file.short_path),
+            name,
+        ).replace("/", "_")
+    )
+    generator_label = "//{}:{}".format(generator_package_directory, name)
```

Comment:
By having a unique package per target, this supports concurrent project generations (assuming each has their own bazel lock).